app/questionnaire/navigator.py

```diff
@@ -209,6 +210,10 @@ def get_blocks(self):
                 for rule in group['routing_rules']:
                     if 'repeat' in rule.keys():
                         no_of_times = evaluate_repeat(rule['repeat'], self.answer_store)
+                        if no_of_times == 0:
```

Comment:
This method seems to doing a lot now. Seems a bit wrong to add a bunch of blocks if they're going to be removed.

What might be better is to default the no_of_times to 1 and override it if there are repeats in the routing rules and have a single loop to iterate over and add them all?